okhttp-tls/src/main/java/okhttp3/tls/HeldCertificate.kt

```diff
@@ -111,52 +111,90 @@ import javax.security.auth.x500.X500Principal
  * a chain of certificates. The server uses a set of trusted root certificates to authenticate the
  * client. Subject alternative names are not used for client authentication.
  */
-class HeldCertificate(private val keyPair: KeyPair, private val certificate: X509Certificate) {
+class HeldCertificate(
+  @get:JvmName("keyPair") val keyPair: KeyPair,
+  @get:JvmName("certificate") val certificate: X509Certificate
+) {
+
+  @JvmName("-deprecated_certificate")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "certificate"),
+      level = DeprecationLevel.WARNING)
   fun certificate(): X509Certificate = certificate
 
+  @JvmName("-deprecated_keyPair")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "keyPair"),
+      level = DeprecationLevel.WARNING)
   fun keyPair(): KeyPair = keyPair
 
   /**
    * Returns the certificate encoded in [PEM format][rfc_7468].
    *
    * [rfc_7468]: https://tools.ietf.org/html/rfc7468
    */
-  fun certificatePem(): String {
-    return buildString {
-      append("-----BEGIN CERTIFICATE-----\n")
-      encodeBase64Lines(certificate.encoded.toByteString())
-      append("-----END CERTIFICATE-----\n")
+  @get:JvmName("certificatePem") val certificatePem: String
```

Comment:
I'm confused whether we want these all to be vals when they do quite a bit of work to construct the result.

Reply:
Doing some work is okay, but the question is really whether this is an intrinsic property of the type or it's a product of the state of the instance. It's probably the latter, and should probably remain as functions.

Reply:
I like that. If it’s acting as a factory it’s not a val.

Reply:
Thanks! Removed the usage of `@get:JvmName(...)` for these factory methods.